### ISSUE TITLE
"Failed to index long" warning in MongodbPanacheTransactionTest

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#37452]
When running `MongodbPanacheTransactionTest`, we end up with the following warning:

```
[INFO] Running io.quarkus.it.mongodb.panache.transaction.MongodbPanacheTransactionTest
2023-12-01 14:26:59,905 WARN  [io.qua.dep.ste.CombinedIndexBuildStep] (build-5) Failed to index long: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: TEST for MongodbPanacheTransactionTest (QuarkusTest)@11cdf948
```

The warning is in fact coming from `IndexWrapper` (I will fix the logger in an upcoming PR) but I think we should try to get rid of it.
I don't know if it's something I should fix in the MongoDB Panache extension or if it's something that we should take care of globally at the Quarkus level.

It could be by indexing all the primitive types? Or ignoring them when trying to index?

/cc @Ladicek @mkouba

**Comment by Ladicek** [comment]
Indexing primitive types makes no sense, there's no class file for them. It would be interesting to know who is attempting to do that, as what they're doing is nonsensical and possibly a bug.

**Comment by Ladicek** [comment]
Oh but that's the on-demand indexing wrapper. Hmm. I'll check.

**Comment by mkouba** [comment]
> Indexing primitive types makes no sense, there are no class files for them. It would be interesting to know who is attempting to do that, as what they're doing is nonsensical and possibly a bug.

:+1: 

> Oh but that's the on-demand indexing wrapper. Hmm. I'll check.

Keep in mind that it's the cloned `IndexWrapper`, not the original from ArC which ignores primitives and arrays.

**Comment by gsmet** [comment]
Yeah, I'm pretty sure the caller is the culprit and should test things a bit more but I'm wondering if we should have a safeguard at some point.

**Comment by Ladicek** [comment]
> Keep in mind that it's the cloned `IndexWrapper`, not the original from ArC which ignores primitives and arrays.

Maybe we should do the same in the Quarkus Core clone. Probably.

**Comment by gsmet** [comment]
@Ladicek while you're at it, can you fix the logger to point to the correct class?
